core/common/src/main/java/alluxio/metrics/Metric.java

```diff
@@ -13,21 +13,29 @@
 
 import com.google.common.base.Objects;
 import com.google.common.base.Preconditions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.io.Serializable;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Map.Entry;
 
 /**
  * A metric of a given instance. The instance can be master, worker, or client.
  */
 public final class Metric implements Serializable {
+  private static final Logger LOG = LoggerFactory.getLogger(Metric.class);
   private static final long serialVersionUID = -2236393414222298333L;
 
   private static final String ID_SEPARATOR = "-id:";
+  public static final String TAG_SEPARATOR = ":";
   private final MetricsSystem.InstanceType mInstanceType;
   private final String mHostname;
   private final String mName;
   private final Long mValue;
   private String mInstanceId;
+  private final Map<String, String> mTags;
```

Comment:
Could we make a `MetricTag` object to make the associations between tag name and value more explicit?

Reply:
I feel it's not necessary for now: tags are simply a map of string to string. I would consider a MetricTag object if it has other functionality added in the future.

Reply:
It would make the code a lot easier to read, could you leave a TODO?

Reply:
I can add a TODO, but I still think it adds unnecessary complexity. For example, the map provides checking existence and replace functionality. But an object of `tag` would require another index data structure for such functionality.